workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -162,6 +170,29 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
     )
   }
 
+  override fun <ResultT> remember(
+    key: String,
+    resultType: KType,
+    vararg inputs: Any?,
+    calculation: () -> ResultT
+  ): ResultT {
+    remembered.forEachStaging {
+      require(key != it.key) {
+        "Expected remember keys to be unique: \"$key\""
+      }
+    }
+
+    val result = remembered.retainOrCreate(
+      predicate = {
+        key == it.key && it.resultType == resultType && inputs.contentEquals(it.inputs)
```

Comment:
Is `contentEquals` on a vararg just working on the array? How expensive is it?

Reply:
The IDE warned me to use it instead of `==`, I don't think that's reliable on `Array`. Not sure where to find the implementations:

```
/**
 * Returns `true` if the two specified arrays are *structurally* equal to one another,
 * i.e. contain the same number of the same elements in the same order.
 * 
 * The elements are compared for equality with the [equals][Any.equals] function.
 * For floating point numbers it means that `NaN` is equal to itself and `-0.0` is not equal to `0.0`.
 */
@SinceKotlin("1.4")
public expect infix fun <T> Array<out T>?.contentEquals(other: Array<out T>?): Boolean
```

Reply:
From the kdoc it looks legit.

Reply:
And yes, the compiler says I'm using the Array version.

Reply:
<img width="529" alt="Screenshot 2025-02-27 at 4 24 09 PM" src="https://github.com/user-attachments/assets/f365b93a-3db2-4d46-8db7-a908cd6d0de7" />